test-utils/src/history_replay.rs

```diff
@@ -0,0 +1,179 @@
+mod history_builder;
```

Comment:
I agree with @cretz, all of this is part of core and should be in the main crate.